src/Nethermind/Nethermind.Network/P2P/Session.cs

```diff
@@ -491,28 +491,32 @@ public void MarkDisconnected(DisconnectReason disconnectReason, DisconnectType d
             if (_context is null)
             {
                 //in case pipeline did not get to p2p - no disconnect delay
-                _channel.DisconnectAsync().ContinueWith(x =>
+                try
```

Comment:
Can split the async part and don't block on it? I'm concerned that this may have unintended consequence.

Reply:
Moved the async out